test/validation/org/apache/cassandra/bridges/ccmbridge/CCMBridge.java

```diff
@@ -176,4 +190,102 @@ private String executeAndRead(String command, Object... args)
             throw new RuntimeException(e);
         }
     }
+
+    public void preserveClusterLogs(String testName){
+
+        int count = 1;
+        String clusterLogsPath = ccmDir + "/" + DEFAULT_CLUSTER_NAME;
+        String folderName = testName;
+        String existingFolder = CASSANDRA_DIR + "/build/test/logs/validation/" + folderName;
+
+        if(checkForFolder(existingFolder))
+        {
+            zipExistingDirectory(existingFolder);
+        }
+
+        while (count <= testNodeCount){
+
+            File sourceFile = new File(clusterLogsPath + "/node" + count + "/logs" + "/system.log");
+
+            File destFile = new File(CASSANDRA_DIR + "/build/test/logs/validation/" + folderName + "/node" + count + ".log");
+
+            try
+            {
+                FileUtils.copyFile(sourceFile, destFile);
+            }
+            catch (IOException e)
+            {
+                e.printStackTrace();
+            }
+
+            count++;
+        }
+
+    }
+
+    public boolean checkForFolder(String dirPath){
+
+        File file = new File(dirPath);
+        return file.exists();
+    }
+
+    public void zipExistingDirectory(String existDir){
+
+        String sourceFolderName =  existDir;
+        long unixTime = System.currentTimeMillis() / 1000L;
+        String outputFileName = existDir + "_" + unixTime + ".zip";
+
+        try
+        {
+            FileOutputStream fos = new FileOutputStream(outputFileName);
+            ZipOutputStream zos = new ZipOutputStream(fos);
+            zos.setLevel(9);
+            compressFiles(zos, sourceFolderName, sourceFolderName);
+            zos.close();
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void compressFiles(ZipOutputStream zos,String folderName,String baseFolderName){
```

Comment:
These last three functions are not specific to CCMBridge. They should be extracted out to a helper class.